planner/cascades/transformation_rules.go

```diff
@@ -594,10 +648,18 @@ func (r *PushSelDownJoin) OnTransform(old *memo.ExprIter) (newExprs []*memo.Grou
 		join.RightJoinKeys = append(join.RightJoinKeys, eqCond.GetArgs()[1].(*expression.Column))
 	}
 	// TODO: Update EqualConditions like what we have done in the method join.updateEQCond() before.
-	leftGroup = buildChildSelectionGroup(sel, leftCond, joinExpr.Children[0])
-	rightGroup = buildChildSelectionGroup(sel, rightCond, joinExpr.Children[1])
+	leftGroup = buildChildSelectionGroup(sel, leftCond, leftGroup)
+	rightGroup = buildChildSelectionGroup(sel, rightCond, rightGroup)
 	newJoinExpr := memo.NewGroupExpr(join)
 	newJoinExpr.SetChildren(leftGroup, rightGroup)
+	if len(remainCond) > 0 {
```

Comment:
One way to avoid `HasBeenPushed` is to check if the `remainCond` deep equals to `sel.Conditions`, if so, we can return a `nil, false, false, nil` here. And it will end the exploration of this GroupExpr.

Reply:
No...we can't. Because we push down `join.OtherCondition` to each side. For example, left outer join, if there is an `on` condition like `t2.a=2`, and `t2` is the inner table. We can push the condition `t2.a is not null` to inner side.

Reply:
Actually, if there is none `where` condition in this case. We can't push it down for the join's `on` conditions. In the future, we should consider the conditions push down for joins.

Reply:
Oh, I see.